File: src/main/java/de/schramm/royalbash/controller/service/core/Context.java
package de.schramm.royalbash.controller.service.core;

import lombok.Builder;
import lombok.Value;

import java.util.Optional;

@Value
@Builder(toBuilder = true)
public class Context {

    private final Game game;
    private final Player owner;
    private final Player targetPlayer;
    private final Spot targetSpot;
    private final Creature targetCreature;

    public Optional<Player> getTargetPlayer() {
        return Optional.ofNullable(targetPlayer);
    }

    public Optional<Spot> getTargetSpot() {
        return Optional.ofNullable(targetSpot);
    }

    public Optional<Creature> getTargetCreature() {
        return Optional.ofNullable(targetCreature);
    }
}

File: src/main/kotlin/de/schramm/royalbash/controller/service/core/Context.kt
package de.schramm.royalbash.controller.service.core

import java.util.*

data class Context(
        val game: Game,
        val owner: Player,
        private val targetPlayer: Player? = null,
        private val targetSpot: Spot? = null,
        private val targetCreature: Creature? = null
) {

    fun getTargetPlayer(): Optional<Player> {
        return Optional.ofNullable(targetPlayer)
    }

    fun getTargetSpot(): Optional<Spot> {
        return Optional.ofNullable(targetSpot)
    }

    fun getTargetCreature(): Optional<Creature> {
        return Optional.ofNullable(targetCreature)
    }
}
